### PR TITLE
Fix clippy warnings. Add nicer error messages for mmap files.

### PR DESCRIPTION
**Pull request by piobab**

### CODE REVIEW

File: src/embedding.rs [hunk 1]
```diff
@@ -191,11 +201,24 @@ where
...
+                    .unwrap_or_else(|_| {
+                        broken_entities.insert(entity_name);
+                    });
             };
         }
 
-        embedding_persistor.finish();
+        if !broken_entities.is_empty() {
+            warn!(
+                "Entities which can't be written to the file: {:?}",
```
Comment (kodieg) [inline src/embedding.rs:214]: Maybe display total count of broken entities and just a few first? Not sure what if it makes sense to display all of them if there is a lot of broken

File: src/embedding.rs [hunk 2]
```diff
@@ -176,8 +177,17 @@ where
         info!("Start saving embeddings.");
 
         let entities_count = self.sparse_matrix_persistor.get_entity_counter();
-        embedding_persistor.put_metadata(entities_count, self.dimension);
+        embedding_persistor
```
Comment (kodieg) [inline src/embedding.rs:180]: Should we proceed further when we could not write metadata? Maybe we should we just panic?

Comment (piobab) [reply src/embedding.rs:180]: You are right. If there is a problem then probably further is the same.

File: src/embedding.rs (hunk 1 above)
Comment (piobab) [reply src/embedding.rs:214]: I will do it. If there are more errors probably there is a bigger issue.

**Review by kodieg — APPROVED** [review]
LGTM

**Pull request merged**